tests/JsonApi/EventListener/FlattenPaginationParametersListenerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\tests\JsonApi\EventListener;
+
+use ApiPlatform\Core\JsonApi\EventListener\FlattenPaginationParametersListener;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ */
+class FlattenPaginationParametersListenerTest extends \PHPUnit_Framework_TestCase
+{
+    private $listener;
+
+    protected function setUp()
+    {
+        $this->listener = new FlattenPaginationParametersListener();
+    }
+
+    public function testOnKernelRequestWithInvalidFormat()
+    {
+        $expectedRequest = new Request();
+        $expectedRequest->setRequestFormat('badformat');
+
+        $request = $expectedRequest->duplicate();
+
+        $eventProphecy = $this->prophesize(GetResponseEvent::class);
+        $eventProphecy->getRequest()->willReturn($request)->shouldBeCalled();
+
+        $this->listener->onKernelRequest($eventProphecy->reveal());
+
+        $this->assertEquals($expectedRequest, $request);
+    }
+
+    public function testOnKernelRequestWithInvalidPage()
+    {
+        $eventProphecy = $this->prophesize(GetResponseEvent::class);
+
+        $expectedRequest = new Request();
+        $expectedRequest->setRequestFormat('jsonapi');
+
+        $request = $expectedRequest->duplicate();
+        $eventProphecy->getRequest()->willReturn($request)->shouldBeCalled();
+        $this->listener->onKernelRequest($eventProphecy->reveal());
+
+        $this->assertEquals($expectedRequest, $request);
+
+        $expectedRequest = $expectedRequest->duplicate(['page' => 'foo']);
+
+        $request = $expectedRequest->duplicate();
+        $eventProphecy->getRequest()->willReturn($request)->shouldBeCalled();
+        $this->listener->onKernelRequest($eventProphecy->reveal());
+
+        $this->assertEquals($expectedRequest, $request);
+    }
+
+    public function testOnKernelRequest()
+    {
+        $request = new Request(['page' => ['size' => 5, 'number' => 3]]);
+        $request->setRequestFormat('jsonapi');
+
+        $eventProphecy = $this->prophesize(GetResponseEvent::class);
+        $eventProphecy->getRequest()->willReturn($request)->shouldBeCalled();
+
+        $this->listener->onKernelRequest($eventProphecy->reveal());
+
+        $expectedRequest = new Request(['page.size' => 5, 'page.number' => 3]);
```

Comment:
we could also test negative page parameters values to avoid a recent issue we had.